packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1555,8 +1555,8 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
             debugPrintStack(label: 'markNeedsLayout() called for $this');
           return true;
         }());
-        owner._nodesNeedingLayout.add(this);
-        owner.requestVisualUpdate();
+        owner!._nodesNeedingLayout.add(this);
+        owner!.requestVisualUpdate();
       }
     }
   }
```

Comment:
i don't understand why markParentNeedsLayout doesn't need `parent` to be nullable